### PR TITLE
allow modifier field to be optional

### PR DESCRIPTION
**Pull request by yads**

### CODE REVIEW

**Comment by dblock** [comment]
Looks good, needs specs, changelog, docs.

**Comment by yads** [comment]
@dblock it looks like the `optional` association option is only available in Mongoid 6. Which this feature needs to work properly on Mongoid 6. Are you aware of how to do feature/version detection on which version of Mongoid this would be running against? I supposed I could also do something like:

```
begin
        belongs_to :modifier, class_name: Mongoid::History.modifier_class_name, optional: true
rescue
        belongs_to :modifier, class_name: Mongoid::History.modifier_class_name
end
```

Do you have any thoughts on how to handle this situation so that this feature works on mongoid versions < 6 and > 6?

**Comment by dblock** [comment]
You can use checks in Mongoid::Compatibility to see what version is being used in both the implementation and the specs, so no `rescue` please.

**Comment by yads** [comment]
@dblock added some specs and using mongoid-compatibility for feature detection. The only thing I had to up the rubocop module size as tracker.rb was right on the cusp of exceeding 176.

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,7 @@ when /3/
 else
   gem 'mongoid', version
 end
+gem 'mongoid-compatibility'
```
Comment (dblock) [inline Gemfile]: This needs to become a .gemspec dependency, otherwise it will be failing at runtime.

Comment (yads) [reply Gemfile]: I believe it's already in the .gemspec file. Is this not it? https://github.com/mongoid/mongoid-history/blob/master/mongoid-history.gemspec#L23

Comment (yads) [reply Gemfile]: Or are you saying to add a version dependency so that it works with version 0.5.1 or above?

Comment (dblock) [reply Gemfile]: Then you don't need to add it here, don't you?

**Comment by yads** [comment]
Yay, finally passing!

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -55,7 +55,7 @@ Metrics/MethodLength:
 # Offense count: 2
 # Configuration parameters: CountComments.
 Metrics/ModuleLength:
-  Max: 176
+  Max: 200
```
Comment (dblock) [inline .rubocop_todo.yml]: Fyi usually we just run `rubocop -a ; rubocop --auto-gen-config`.

**Comment by dblock** [comment]
Merging this, thanks!

File: .rubocop_todo.yml (hunk 2 above)
Comment (yads) [reply .rubocop_todo.yml]: Ah ok, I was wondering why 176 was the old value :)

**Comment by dblock** [comment]
Coming from #221. 

Is the modifier still not actually enforced in Mongoid 6? A spec like https://github.com/mongoid/mongoid-history/blob/master/spec/unit/attributes/destroy_spec.rb never uses a modifier but succeeds in 6 and now fails in 7 saying a modifier is required.

**Pull request merged**